pkg/system/ipvs.go

```diff
@@ -73,7 +75,38 @@ func (i *ipvs) Get() ([]string, error) {
 	buf := bytes.NewBuffer(stdout)
 	scanner := bufio.NewScanner(buf)
 	for scanner.Scan() {
-		out = append(out, scanner.Text())
+		rule := scanner.Text()
+		// filter away v6 rules
+		if !strings.Contains(rule, "[") && !strings.Contains(rule, "]") {
+			out = append(out, rule)
+		}
+	}
+
+	return out, nil
+}
+
+// getConfiguredIPVS returns the output of `ipvsadm -Sn`
+// That IPVS command returns a list of director VIP addresses sorted in lexicographic order by address:port,
+// with backends sorted by realserver address:port.
+// GetV6 filters only ipv6 rules. Sadly there is no native ipvsadm command to filter this
+func (i *ipvs) GetV6() ([]string, error) {
+
+	// run the ipvsadm command
+	cmd := exec.CommandContext(i.ctx, "ipvsadm", "-Sn")
+	stdout, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("ipvsadm -Sn failed with %v", err)
+	}
+
+	out := []string{}
+	buf := bytes.NewBuffer(stdout)
+	scanner := bufio.NewScanner(buf)
+	for scanner.Scan() {
+		rule := scanner.Text()
+		// filter only v6 rules
+		if strings.Contains(rule, "[") && strings.Contains(rule, "]") {
```

Comment:
defend yourself! some comment about `cat /proc/net/ip_vs` might be valuable for the future


lmao http://svn.linuxvirtualserver.org/repos/ipvsadm/trunk/ipvsadm.c
```
	if (buf[0] == '[') {
		buf++;
		portp = strchr(buf, ']');
		if (portp == NULL)
			return SERVICE_NONE;
		*portp = '\0';
		portp++;
		if (*portp == ':')
			*portp = '\0';
		else
			return SERVICE_NONE;
	}
```

Reply:
seems like searching for brackets is totally normal